File: app/src/test/java/com/abishov/hexocat/Inject.java
package com.abishov.hexocat;

import com.abishov.hexocat.common.network.HexocatAdapterFactory;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

/**
 * Dependency provider for unit tests.
 */
public final class Inject {

  private Inject() {
    // no instances
  }

  public static Gson gson() {
    return new GsonBuilder()
        .registerTypeAdapterFactory(HexocatAdapterFactory.Companion.create())
        .create();
  }
}

File: app/src/test/java/com/abishov/hexocat/Inject.kt
package com.abishov.hexocat

import com.squareup.moshi.Moshi
import com.squareup.moshi.kotlin.KotlinJsonAdapterFactory

object Inject {

  fun moshi(): Moshi {
    return Moshi.Builder()
      .add(KotlinJsonAdapterFactory())
      .build()
  }
}
